_posts/2020-08-05-using-open-policy-agent-with-strimzi-and-apache-kafka.md

```diff
@@ -0,0 +1,456 @@
+It is also supported in the User Operator, so you can configure the ACL rules directly in the `KafkaUser` custom resources.
+So you can easily configure everything in YAML, store it in your Git repository and use GitOps to manage the resources.
+
+## Keycloak Authorization Services
+
+[Keycloak](https://www.keycloak.org/) is an open source identity and access management server.
+Strimzi supports authentication of Kafka clients using OAuth 2.0 access tokens.
+And Keycloak is one of the servers you can use for it (see our older blog post _[Kafka authentication using OAuth 2.0](https://strimzi.io/blog/2019/10/25/kafka-authentication-using-oauth-2.0/)_).
+But you can also use it for authorization thanks to our own `KeycloakRBACAuthorizer`.
+
+The `keycloak` authorization is based on the [Keycloak Authorization Services](https://www.keycloak.org/docs/latest/authorization_services/index.html).
+It can be used only in combination with OAuth 2.0 authentication.
+Security policies and permissions are defined in Keycloak and are used to grant access to resources on Kafka brokers. 
+Users and clients are matched against policies that permit access to perform specific actions on Kafka brokers.
+You can also use the _federation_ feature of Keycloak to map the policies for example against LDAP groups.
+The `KeycloakRBACAuthorizer` will fetch the list of granted permissions from Keycloak and enforce the authorization rules locally to make sure it doesn't negatively impact performance.
+
+The main advantage of Keycloak is that you can have everything managed centrally.
+Not just for Strimzi and Apache Kafka, but for all your applications which support OAuth.
+It can also handle both authentication and authorization.
+So you do not need to have different servers for authentication and authorization and all data are stored in one place.
+
+On the other hand, you will still need to run and manage Keycloak.
+So unless you already use it or at least plan to use some central authentication / authorization server anyway, this will be additional effort you would need to deal with.
+
+## Open Policy Agent
+
+[Open Policy Agent](https://www.openpolicyagent.org/) (OPA) is an open source policy engine.
+It can be used to enforce policies across your whole architecture stack.
+The policies are declarative and to write them it is using its own language called _Rego_.
+OPA can be used for policy based control of many different applications.
+It can be used with Kubernetes, APIs, SSH, CI/CD pipelines and many more.
+
+![Open Policy Agent](/assets/images/posts/2020-08-05-using-open-policy-agent-with-strimzi-and-apache-kafka.png)
+
+OPA decouples the policy decision making from enforcing the decision.
+When used with Kafka, the authorizer running inside Kafka will call OPA server to evaluate the policy based on the input from the authorizer.
+The input will be the same set of information as with any other Kafka authorizer.
+It is described in the introduction to this blog post.
```

Comment:
Maybe title the start so we can link to it again for convenience

Reply:
The blog posts do not support deep linking. So I do not think it is needed.